tests/main/views/test_buyers.py

```diff
@@ -3676,3 +3645,125 @@ def test_award_brief_details_post_raises_400_and_displays_error_messages_and_pre
+            lot_slug="digital-outcomes",
+            status='closed'
+        )['briefs']
+        self.data_api_client.get_brief.return_value = {"briefs": self.brief}
+        self.login_as_buyer()
+
+    def teardown_method(self, method):
+        self.data_api_client_patch.stop()
+        super(TestCancelBrief, self).teardown_method(method)
+
+    def test_cancel_brief_200s_with_correct_default_content(self):
+        self.login_as_buyer()
+        res = self.client.get(self.url.format(brief_id=1234))
+
+        assert res.status_code == 200
+        document = html.fromstring(res.get_data(as_text=True))
+        self.assert_breadcrumbs(res, extra_breadcrumbs=[
+            (
+                'I need a thing to do a thing',
+                '/buyers/frameworks/digital-outcomes-and-specialists-2/requirements/digital-outcomes/1234'
+            )
+        ])
+
+        page_title = document.xpath('//h1')[0].text_content()
+        assert "Why do you need to cancel {}?".format(self.brief.get('title')) in page_title
+
+        submit_button = document.xpath('//input[@class="button-save" and @value="Update requirement"]')
+        assert len(submit_button) == 1
+
+    def test_404_if_user_is_not_brief_owner(self):
+        self.data_api_client.get_brief.return_value['briefs']['users'][0]['id'] = 234
+
+        res = self.client.get(self.url.format(brief_id=self.brief["id"]))
+
+        assert res.status_code == 404
+
+    @pytest.mark.parametrize('status', ['withdrawn', 'draft', 'live', 'cancelled', 'unsuccessful', 'awarded'])
+    def test_404_if_brief_not_closed(self, status):
+        self.data_api_client.get_brief.return_value['briefs']['status'] = status
+
+        res = self.client.get(self.url.format(brief_id=self.brief["id"]))
+
+        assert res.status_code == 404
+
+    @pytest.mark.parametrize('framework_status', ['live', 'expired'])
+    def test_200_for_acceptable_framework_statuses(self, framework_status):
+        self.data_api_client.get_framework.return_value['frameworks']['status'] = framework_status
+
+        res = self.client.get(self.url.format(brief_id=123))
+
+        assert res.status_code == 200
+
+    def test_that_no_option_chosen_triggers_error(self):
+        res = self.client.post(self.url.format(brief_id=123))
+
+        document = html.fromstring(res.get_data(as_text=True))
+        validation_message = document.xpath('//span[@class="validation-message"]')[0].text_content()
+
+        assert res.status_code == 400
+        assert "You need to answer this question." in validation_message
+        assert self.data_api_client.cancel_brief.called is False
+
+    def test_cancel_triggers_cancel_brief(self):
+        res = self.client.post(
+            self.url.format(brief_id=123), data={'cancel_reason': 'cancel'}
+        )
+
+        assert res.status_code == 302
+        self.data_api_client.cancel_brief.assert_called_once_with('123', user='buyer@email.com')
+        self.assert_flashes("updated-brief")
```

Comment:
It feels like this assertion should live in the test that checks the redirect as they are both to do with what should be shown to the user next

Reply:
Also ideally we would be asserting both the key and the value of the flash message we are setting

Reply:
Done 👍